src/Sulu/Bundle/AdminBundle/Resources/js/components/RectangleSelection/RectangleSelection.js

```diff
@@ -1,91 +1,157 @@
-import React from 'react';
 import withContainerSize from '../withContainerSize';
 import type {Normalizer, RectangleChange, SelectionData} from './types';
 import ModifiableRectangle from './ModifiableRectangle';
 import PositionNormalizer from './normalizers/PositionNormalizer';
 import RatioNormalizer from './normalizers/RatioNormalizer';
 import RoundingNormalizer from './normalizers/RoundingNormalizer';
 import SizeNormalizer from './normalizers/SizeNormalizer';
+import withPercentageValues from './withPercentageValues';
 import rectangleSelectionStyles from './rectangleSelection.scss';
 
 type Props = {
+    backdrop: boolean,
     children?: Node,
     containerHeight: number,
     containerWidth: number,
-    minHeight?: number,
-    minWidth?: number,
+    disabled: boolean,
+    label?: string,
+    minHeight: number | typeof undefined,
+    minSizeNotification: boolean,
+    minWidth: number | typeof undefined,
     onChange: (s: ?SelectionData) => void,
+    onFinish?: () => void,
     round: boolean,
+    usePercentageValues: boolean,
     value: SelectionData | typeof undefined,
 };
 
 @observer
-class RectangleSelection extends React.Component<Props> {
+class RectangleSelectionComponent extends React.Component<Props> {
     static defaultProps = {
+        backdrop: true,
+        disabled: false,
+        minHeight: undefined,
+        minSizeNotification: true,
+        minWidth: undefined,
         round: true,
+        usePercentageValues: false,
+    };
+
+    @computed get value() {
+        const {value} = this.props;
+
+        if (!value) {
+            return this.maximumSelection;
+        }
+
+        return value;
+    }
+
+    componentDidMount() {
+        this.setInitialValue();
+    }
+
+    componentDidUpdate() {
+        this.setInitialValue();
+    }
+
+    setInitialValue = () => {
+        const {onChange, value} = this.props;
+
+        if (!this.props.containerHeight || !this.props.containerWidth) {
+            return;
+        }
+
+        if (!value) {
+            onChange(this.value);
+        }
     };
 
     static createNormalizers(props: Props): Array<Normalizer> {
-        if (!props.containerWidth || !props.containerHeight) {
+        const {
+            containerWidth,
+            containerHeight,
+            minWidth,
+            minHeight,
+            round,
+        } = props;
+
+        if (!containerWidth || !containerHeight) {
             return [];
         }
 
-        const normalizers = [];
-
-        normalizers.push(new SizeNormalizer(
-            props.containerWidth,
-            props.containerHeight,
-            props.minWidth,
-            props.minHeight
-        ));
-
-        if (props.minWidth && props.minHeight) {
-            normalizers.push(new RatioNormalizer(
-                props.containerWidth,
-                props.containerHeight,
-                props.minWidth,
-                props.minHeight
-            ));
-        }
+        let normalizers = [
```

Comment:
Why is that not a `const` anymore? From what I can see this could be achieved the same way as before, without creating so many arrays, by just leaving this variable as `const` and push new values.